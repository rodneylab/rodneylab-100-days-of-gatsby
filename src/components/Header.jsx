import React from 'react';
import { Box, Heading } from '@chakra-ui/react';
import { graphql, Link, useStaticQuery } from 'gatsby';
import PropTypes from 'prop-types';

import { DESKTOP_BREAKPOINT } from '../constants/sizes';
import { useMediaQuery } from '../hooks';

const ListLink = ({
  ariaLabel, to, children: listLinkChildren, isDesktop,
}) => (
  <li style={isDesktop ? {} : { padding: '15px 0px' }}>
    <Link aria-label={ariaLabel} to={to}>
      {listLinkChildren}
    </Link>
  </li>
);

ListLink.defaultProps = {
  isDesktop: false,
};

ListLink.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  isDesktop: PropTypes.bool,
};

const sortLocation = (a, b) => {
  if (a.city < b.city) {
    return -1;
  }
  if (a.city > b.city) {
    return 1;
  }
  return 0;
};

export const PureHeader = ({ data }) => {
  const isDesktop = useMediaQuery(`(min-width: ${DESKTOP_BREAKPOINT}px)`);

  return (
    <header>
      {/* <Box bg="blue.800" color="yellow.500"> */}
      <Box bg="blue.700" color="white">
        <Heading>AudioC0RE - headphones sharing</Heading>
        <nav>
          <ul>
            <ListLink ariaLabel="Open Home page" to="/home" isDesktop={isDesktop}>
              Home
            </ListLink>
            <ListLink ariaLabel="Open About page" to="/about" isDesktop={isDesktop}>
              About
            </ListLink>
            <ListLink ariaLabel="Open Location page" to="/location" isDesktop={isDesktop}>
              Locations:
            </ListLink>
            <li>
              <ul>
                {data.allContentfulLocation.nodes.sort(sortLocation).map((location) => {
                  const { city, slug } = location;
                  return (
                    <ListLink
                      ariaLabel={`Open ${city} location page`}
                      to={`/location/${slug}`}
                      key={slug}
                      isDesktop={isDesktop}
                    >
                      {city}
                    </ListLink>
                  );
                })}
              </ul>
            </li>
          </ul>
        </nav>
      </Box>
    </header>
  );
};

PureHeader.propTypes = {
  data: PropTypes.shape({
    allContentfulLocation: PropTypes.shape({
      nodes: PropTypes.arrayOf(
        PropTypes.shape({
          city: PropTypes.string,
          slug: PropTypes.string,
        }),
      ),
    }),
  }).isRequired,
};

export const query = graphql`
  fragment HeaderFragment on ContentfulLocationConnection {
    nodes {
      city
      slug
    }
  }
`;

const Header = () => {
  const data = useStaticQuery(
    graphql`
      query HeaderQuery {
        allContentfulLocation {
          ...HeaderFragment
        }
      }
    `,
  );
  return <PureHeader data={data} />;
};

export { Header as default };
