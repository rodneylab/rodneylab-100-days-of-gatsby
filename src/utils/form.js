import { createClient } from '@formium/client';

export const formium = createClient(process.env.GATSBY_FORMIUM_PROJECTID);

export const validate = (values) => {
  const errors = {};
  if (!values.name) {
    errors.name = 'Please enter your name.';
  }
  if (!values.email) {
    errors.email = 'Please enter your email address.';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Please check your email address.';
  }
  if (!values.message) {
    errors.message = 'Please enter a message.';
  } else if (values.message.length > 1024) {
    errors.message = 'Please limit your message to 1000 characters.';
  }
  return errors;
};
