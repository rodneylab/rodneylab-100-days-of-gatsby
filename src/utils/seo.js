import { getSrc } from 'gatsby-plugin-image';

export const getFeaturedImage = ({ featuredImage }) => ({
  url: getSrc(featuredImage.localFile),
  width: 992,
  height: 730,
});

export const getOgImage = ({ ogImage }) => ({
  url: ogImage ? getSrc(ogImage.localFile) : null,
  width: 1200,
  height: 627,
});

export const getOgSquareImage = ({ ogSquareImage }) => ({
  url: ogSquareImage ? getSrc(ogSquareImage.localFile) : null,
  width: 400,
  height: 400,
});

export const getTwitterImage = ({ twitterImage }) => ({
  url: twitterImage ? getSrc(twitterImage.localFile) : null,
  width: 800,
  height: 418,
});
