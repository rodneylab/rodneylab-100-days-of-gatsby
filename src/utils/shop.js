const formatPrice = (currency, value) => Intl.NumberFormat(undefined, {
  currency,
  minimumFractionDigits: 2,
  style: 'currency',
}).format(value);

export { formatPrice as default };

export const localStorageKey = 'a856f4';
