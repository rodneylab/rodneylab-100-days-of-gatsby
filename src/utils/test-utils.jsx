import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { createMemorySource, createHistory, LocationProvider } from '@reach/router';

const source = createMemorySource('active');
const history = createHistory(source);

const locationProvider = ({ children }) => (
  <LocationProvider history={history}>{children}</LocationProvider>
);

export const customRender = (ui, options) => render(ui, { wrapper: locationProvider, ...options });

export const getMetaByName = (metaName) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('name') === metaName) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};

export const getMetaByProperty = (metaProperty) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('property') === metaProperty) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};
