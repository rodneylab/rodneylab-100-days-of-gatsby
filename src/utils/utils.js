// eslint-disable-next-line import/prefer-default-export
export const isBrowser = typeof window !== 'undefined';
