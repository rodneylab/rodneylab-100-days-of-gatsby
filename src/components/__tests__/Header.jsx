import React from 'react';

import data from './__fixtures__/header';
import { PureHeader as Header } from '../Header';
import { customRender } from '../../utils/test-utils';

beforeEach(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
});

describe('Header', () => {
  it('renders correctly', async () => {
    const elementToRender = <Header data={data} />;
    const { asFragment } = customRender(elementToRender);
    expect(asFragment(elementToRender)).toMatchSnapshot();
  });
});
