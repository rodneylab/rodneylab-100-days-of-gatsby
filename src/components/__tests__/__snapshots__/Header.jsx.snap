// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header>
    <h1>
      AudioC0RE - headphones sharing
    </h1>
    <nav>
      <ul>
        <li
          style="padding: 15px 0px;"
        >
          <a
            aria-label="Open Home page"
            href="/home"
          >
            Home
          </a>
        </li>
        <li
          style="padding: 15px 0px;"
        >
          <a
            aria-label="Open About page"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          style="padding: 15px 0px;"
        >
          <a
            aria-label="Open Location page"
            href="/location"
          >
            Locations:
          </a>
        </li>
        <li>
          <ul>
            <li
              style="padding: 15px 0px;"
            >
              <a
                aria-label="Open Cape Town location page"
                href="/location/cape-town"
              >
                Cape Town
              </a>
            </li>
            <li
              style="padding: 15px 0px;"
            >
              <a
                aria-label="Open London location page"
                href="/location/london"
              >
                London
              </a>
            </li>
            <li
              style="padding: 15px 0px;"
            >
              <a
                aria-label="Open Rio de Janeiro location page"
                href="/location/rio-de-janeiro"
              >
                Rio de Janeiro
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
